src/main/java/com/puppycrawl/tools/checkstyle/GlobalStatefulCheck.java

```diff
@@ -38,7 +38,8 @@
  * This is similar to multi-file validation, which checkstyle does not support fully yet.
  * Please refer to https://github.com/checkstyle/checkstyle/issues/3540 for details.
  *
- * @noinspection AnnotationClass, ClassIndependentOfModule, unused, ClassOnlyUsedInOnePackage
+ * @noinspection AnnotationClass, ClassIndependentOfModule, ClassOnlyUsedInOnePackage
+ * @noinspectionreason we support only Java 11+, we keep this annotation at top level by design
```

Comment:
I am not following how this inspection and in other files relate to Java <=10 .

Reply:
Annotation classes are not supported on older versions of Java. (pre Java 6 I think). We should probably just not even use this inspection anymore.

Reply:
Are we talking about `@interface TypeAnnotation {}` or something else?

Reply:
> Are we talking about @interface TypeAnnotation {} or something else?

Yes, that would be an example of an Annotation class.

Reply:
I am good.